templates/overrides/navigation/block--system-branding-block.html.twig

```diff
@@ -13,44 +12,46 @@
  * - site_slogan: Slogan for site as defined in Site information settings.
  */
 #}
-{%
-  set logo_classes = [
-    'text-decoration-none',
-    'site-logo',
-  ]
-%}
-{%
-  set name_classes = [
-    'text-decoration-none',
-    'align-bottom',
-    bcl_header_style == 'light' ? 'text-dark': 'text-white',
-  ]
-%}
-{%
-  set img_classes = [
-    'd-none',
-    'd-lg-inline-block',
-  ]
-%}
-
-{% set site_logo_attr = create_attribute() %}
-{% set site_logo_img_attr = create_attribute() %}
-{% set site_name_attr = create_attribute() %}
+{% set light = bcl_header_style == 'light' %}
 
 {% block content %}
-  {% if site_logo %}
-    <a {{ site_logo_attr.addClass(logo_classes).setAttribute('rel', 'Home') }} href="{{ path('<front>') }}">
-      <img {{ site_logo_img_attr.addClass(img_classes).setAttribute('alt', 'Home'|t) }} src="{{ site_logo }}" />
-    </a>
-  {% endif %}
-  {% if site_name %}
-    <div class="site-name bcl-header__site-name">
-      <a {{ site_name_attr.addClass(name_classes).setAttribute('rel', 'Home') }} href="{{ path('<front>') }}">
-        {{ site_name }}
-      </a>
-    </div>
-  {% endif %}
-  {% if site_slogan %}
-    <div class="site-slogan">{{ site_slogan }}</div>
-  {% endif %}
+  {% apply spaceless %}
+    {# Project name. #}
+    {% if site_logo is not empty or site_name is not empty %}
+      {% set project_classes = 'bcl-header__project' %}
+      {% if light is not empty %}
+        {% set project_classes = project_classes ~ ' light' %}
+      {% endif %}
+      <div {{ create_attribute().addClass(project_classes) }}>
+        <div class="container">
+          {% if site_logo is not empty and bcl_component_library != 'neutral' %}
+            {% set img_classes = ['d-none','d-lg-inline-block', 'text-decoration-none', 'site-logo'] %}
+            <a {{ create_attribute().addClass(img_classes).setAttribute('rel', 'Home') }} href="{{ path('<front>') }}">
+              <img {{ site_logo_img_attr.setAttribute('alt', 'Home'|t) }} src="{{ site_logo }}"/>
+            </a>
+          {% endif %}
+          {% if site_name is not empty %}
+            {% set name_classes = [
+              'text-decoration-none',
+              'align-bottom',
+              light ? 'text-dark': 'text-white',
```

Comment:
something like this needs to be added also to the title printed in mobile, see
<img width="311" alt="Screenshot 2022-02-23 at 18 58 24" src="https://user-images.githubusercontent.com/11507647/155368434-61b9aaf5-0735-46ce-b635-de24cc5f54d2.png">
(I also see a strange line there)